app/Models/AccessControlledModel.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Log;
+
+use App\Repositories\RolesRepository as Repo;
+
+class AccessControlledModel extends Model
+{
+
+    // Needs to be false for seeding the database.
+    protected static $ACCESS_CONTROL_ENABLED = true;
+
+    private $roles = array();
```

Comment:
There currently is ambiguity between roles and getRoles() on Models and AccessControlledModels. This should be resolved